crates/core/src/test_scenario.rs

```diff
@@ -182,11 +190,22 @@ where
                     .expect("failed to estimate gas");
 
                 // inject missing fields into tx_req.tx
-                let tx = tx_req
+                let mut tx = tx_req
                     .tx
-                    .with_gas_price(gas_price)
-                    .with_chain_id(chain_id)
-                    .with_gas_limit(gas_limit);
+                    .with_chain_id(chain_id);
+
+                tx = match tx_type {
+                    TxType::Legacy => {
+                        tx.with_gas_price(gas_price).with_gas_limit(gas_limit)
+                    }
+                    TxType::Eip1559 => {
+                        tx
+                            .with_max_fee_per_gas(gas_price + (gas_price / 5))
+                            .with_max_priority_fee_per_gas(gas_price)
+                            .with_gas_limit(gas_limit + (gas_limit / 6))
+                    }
+                    _ => tx,
```

Comment:
If `tx_type` doesn't match one of these branches, won't the tx be lacking gas params? Feels like we should have a default gas params assignment in the fallback branch.